src/messaging/ReliableMessageMgr.cpp

```diff
@@ -101,6 +101,27 @@ void ReliableMessageMgr::TicklessDebugDumpRetransTable(const char * log)
 #endif
 }
 
+void ReliableMessageMgr::TransmitEventAnalyticNotification(const RetransTableEntry & entry, const SessionHandle & sessionHandle,
+                                                           const ReliableMessageAnalyticsDelegate::EventType & eventType)
+{
+    if (!mAnalyticsDelegate)
+    {
+        return;
+    }
+    uint32_t messageCounter = entry.retainedBuf.GetMessageCounter();
+    auto fabricIndex        = sessionHandle->GetFabricIndex();
+    auto destination        = kUndefinedNodeId;
```

Comment:
Consider recording the transport type and also the time that had been awaited before a retransmit, for retransmits.

Reply:
Is it okay if I add those in a follow up PR? I am trying to figure out plumbing of this existing data first. I intentionally bade the callback use `struct TransmitEvent` as the parameter to lend itself to expanding to include other data

Reply:
Disccused offline that follow up is acceptable